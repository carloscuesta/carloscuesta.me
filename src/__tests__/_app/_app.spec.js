import renderer from 'react-test-renderer'
import Router from 'next/router'

import App from 'src/pages/_app.js'
import * as stubs from './stubs'

jest.mock('next-seo', () => ({ DefaultSeo: 'DefaultSeo' }))

jest.mock('next/router', () => ({
  pathname: '',
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn()
  }
}))

Router.useRouter = () => ({
  pathname: '',
  events: {
    off: jest.fn(),
    on: jest.fn(),
    emit: jest.fn()
  }
})

/*
  I'm mocking the components since those are unit tested.
  So there's no point to test the rendering of them twice.
  It's more easier to manage little snapshots than big ones.
*/

jest.mock('src/components/shared/Layout', () => 'Layout')

describe('_app', () => {
  describe('page', () => {
    it('should match App page', () => {
      const wrapper = renderer.create(
        <App {...stubs.props} />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should subscribe on mount to Router.events routeChangeComplete', () => {
      renderer.create(
        <App {...stubs.props} />
      )

      expect(Router.events.on).toHaveBeenCalledWith(
        'routeChangeComplete',
        expect.any(Function)
      )
    })

    it('should unsubscribe on unmount to Router.events routeChangeComplete', () => {
      const wrapper = renderer.create(
        <App {...stubs.props} />
      )

      wrapper.unmount()

      expect(Router.events.off).toHaveBeenCalledWith(
        'routeChangeComplete',
        expect.any(Function)
      )
    })
  })
})
