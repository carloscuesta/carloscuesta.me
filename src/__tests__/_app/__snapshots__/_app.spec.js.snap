// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_app page should match App page 1`] = `
<Layout>
  <DefaultSeo
    description="I'm a Front End engineer based in Barcelona, Spain. Coding addicted that is focused on JavaScript. I like design, technology, sneakers and I ❤️ OpenSource."
    openGraph={
      Object {
        "locale": "en_GB",
        "site_name": "Carlos Cuesta",
        "type": "website",
        "url": "https://carloscuesta.me",
      }
    }
    title="Carlos Cuesta – Front End Engineer"
    twitter={
      Object {
        "cardType": "summary_large_image",
        "handle": "@crloscuesta",
        "site": "@crloscuesta",
      }
    }
  />
  <div
    test=""
  >
    Component
  </div>
</Layout>
`;
